examples/counter-next/shared/wit/world.wit

```diff
@@ -0,0 +1,7 @@
+package counter-next:shared;
+
+world shared {
+    export update: func(data: list<u8>) -> list<u8>;
+    export resolve: func(effect-id: u32, data: list<u8>) -> list<u8>;
+    export view: func() -> list<u8>;
+}
```

Comment:
Not saying to do this now, but to align with the other FFIs, we could treat the core instance as a resource owned by the component, which the host creates using a function like `new`. 

That would then allow use of middleware (I think 🤔 ) and also multiple instances.

Reply:
I started there, and then wound it back to something simpler for now. But deffo worth exploring once we prove this bit out.